Xamarin.PropertyEditing/ViewModels/PropertiesViewModel.cs

```diff
@@ -117,8 +202,44 @@ private void RemoveProperties (IEnumerable<PropertyViewModel> properties)
 			OnRemoveProperties (properties);
 		}
 
+		private async void SetObjectName (string value)
+		{
+			if (this.nameable == null)
+				return;
+
+			try {
+				await this.nameable.SetNameAsync (value);
```

Comment:
Is there an ordering guarantee hidden in here to ensure the correct final value is used if SetObjectName is invoked multiple times?

Reply:
No, it's up to the editor to guarantee (which happens for free if it's not actually async). It's likely that this will move when we introduce the async UI mechanics.